cla-backend/cla/controllers/signing.py

```diff
@@ -207,6 +211,30 @@ def return_url(signature_id, event=None):  # pylint: disable=unused-argument
     ret_url = signature.get_signature_return_url()
     if ret_url is not None:
         cla.log.info('Signature success - sending user to return_url: %s', ret_url)
+        try:
+            project = get_project_instance()
+            project.load(str(signature.get_signature_project_id()))
+        except DoesNotExist as err:
+            cla.log.error('Invalid project_id provided when trying to send user back to'\
+                        'their return_url : %s', signature.get_signature_project_id())
+
+        if project.get_version() == 'v2':
+            if signature.get_signature_reference_type() == 'company':
+                cla.log.info('Getting company instance : %s ', signature.get_signature_reference_id())
+                try:
+                    company = get_company_instance()
+                    company.load(str(signature.get_signature_reference_id()))
+                except DoesNotExist as err:
+                    cla.log.error('Invalid company_id provided : err: %s', signature.get_signature_reference_id)
+                user_service = UserService
+                cla.log.info('Checking if cla managers have cla-manager role permission')
+                while True:
```

Comment:
Should we add a delay with a maximum number of attempts?

Reply:
Sleep 0.5 seconds, retry up to N times?

Reply:
So what would be the preferred number ?